File: samples/WebFormsSampleVB/Templates/RazorTemplate.vb
Imports Microsoft.AspNetCore.Hosting
Imports My.AspNetCore.WebForms.Templating
Imports RazorLight
Imports RazorLight.Extensions

Namespace WebFormsSampleVB.Templates
    Public Class RazorTemplate
        Inherits Template
        Private _hostingEnvironment As IHostingEnvironment

        Public Sub New(ByVal hostingEnvironment As IHostingEnvironment)
            _hostingEnvironment = hostingEnvironment
        End Sub

        Public Overrides Function Parse(ByVal template As String, ByVal model As Object) As String
            Dim root = _hostingEnvironment.WebRootPath
            Dim engine = EngineFactory.CreatePhysical(root)
            Dim result = engine.ParseString(template,
                New With {.ApplicationName = "ASP.NET Core Web Forms", .Title = model.Title})

            Return result
        End Function
    End Class
End Namespace

File: samples/WebFormsSampleCS/Templates/RazorTemplate.cs
using Microsoft.AspNetCore.Hosting;
using My.AspNetCore.WebForms.Templating;
using RazorLight;
using RazorLight.Extensions;

namespace WebFormsSampleCS.Templates
{
    public class RazorTemplate : Template
    {
        private IHostingEnvironment _hostingEnvironment;

        public RazorTemplate(IHostingEnvironment hostingEnvironment)
        {
            _hostingEnvironment = hostingEnvironment;
        }

        public override string Parse(string template, dynamic model)
        {
            var root = _hostingEnvironment.WebRootPath;
            var engine = EngineFactory.CreatePhysical(root);
            var result = engine.ParseString(template,
                new { ApplicationName = "ASP.NET Core Web Forms", Title = model.Title });

            return result;
        }
    }
}
